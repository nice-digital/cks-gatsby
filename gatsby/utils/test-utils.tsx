import React, { ReactElement } from "react";
import { render, RenderResult } from "@testing-library/react";
import {
	createMemorySource,
	createHistory,
	LocationProvider,
	History,
} from "@reach/router";

// Util for rendering a component wrapped within Reach Router's LocationProvider
// See https://testing-library.com/docs/example-reach-router
export const renderWithRouter = (
	ui: ReactElement,
	{ route = "/", history = createHistory(createMemorySource(route)) } = {}
): { history: History } & RenderResult => ({
	...render(<LocationProvider history={history}>{ui}</LocationProvider>),
	// adding `history` to the returned utilities to allow us
	// to reference it in our tests (just try to avoid using
	// this to test implementation details).
	history,
});
