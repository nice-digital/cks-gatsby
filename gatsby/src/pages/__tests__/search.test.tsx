import React from "react";
import { enableFetchMocks } from "jest-fetch-mock";
import { renderWithRouter } from "test-utils";

import SearchPage from "../search";

enableFetchMocks();

describe("Search Page", () => {
	it("should render a loading message before the request comes in", () => {
		const el = renderWithRouter(<SearchPage />);
		expect(el.getByText("loading")).toBeInTheDocument();
	});
});
