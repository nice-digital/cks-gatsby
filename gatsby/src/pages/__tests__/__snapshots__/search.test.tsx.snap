// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page should render a loading message before the request comes in 1`] = `
<div>
  <main
    class="container"
    id="content-start"
  >
    <div
      class="page-header"
    >
      <h1
        class="page-header__heading"
      >
        Search results
      </h1>
      <p
        class="page-header__lead"
      >
        loading
      </p>
    </div>
  </main>
</div>
`;
